### PR TITLE
added support for SK6812 to WS2812 device

### PR DESCRIPTION
**Pull request by SoftTacos**
Adds support for SK6812 (RGBA) LEDs to WS2812 device.

PR for issue: https://github.com/tinygo-org/drivers/issues/608

### CODE REVIEW

File: ws2812/ws2812.go [hunk 1]
```diff
@@ -12,15 +12,27 @@ import (
 
 var errUnknownClockSpeed = errors.New("ws2812: unknown CPU clock speed")
 
+type DeviceType uint8
+
+const (
+	WS2812 DeviceType = iota // RGB
+	SK6812                   // RGBA / RGBW
+)
```
Comment (SoftTacos) [inline ws2812/ws2812.go]: If we would rather opt for making an entirely separate SK6812 device and leverage the same calls that Deivce.WriteByte uses, I'm happy to do that too.

**Comment by deadprogram** [comment]
Probably creating specific constructors vs. modifying the current one would be a better approach.

```go
// deprecated, use NewWS2812 or NewSK6812 depending on which device you want.
// calls NewWS2812() to avoid breaking everyone's existing code.
func New(pin machine.Pin) Device

// returns WS2812
func NewWS2812(pin machine.Pin) Device

// returns SK6812
func NewSK6812(pin machine.Pin) Device
```

**Comment by SoftTacos** [comment]
Love it, I'll get that changed soon.

**Comment by deadprogram** [comment]
Thanks for working on this @SoftTacos now squash/merging.

**Pull request merged**